app/src/androidTest/java/ch/hikemate/app/ui/map/RunHikeScreenTest.kt

```diff
@@ -1,145 +1,228 @@
 import ch.hikemate.app.ui.components.DetailRow
 import ch.hikemate.app.ui.navigation.NavigationActions
-import com.google.firebase.Timestamp
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.test.UnconfinedTestDispatcher
 import kotlinx.coroutines.test.runTest
+import org.junit.Assert.fail
 import org.junit.Before
 import org.junit.Rule
 import org.junit.Test
 import org.mockito.Mockito.doNothing
 import org.mockito.Mockito.mock
 import org.mockito.Mockito.`when`
 import org.mockito.kotlin.any
-import org.mockito.kotlin.eq
 import org.mockito.kotlin.verify
 
+@OptIn(ExperimentalCoroutinesApi::class)
 class RunHikeScreenTest {
 
   @get:Rule val composeTestRule = createComposeRule()
 
   private lateinit var mockNavigationActions: NavigationActions
-  private lateinit var profileRepository: ProfileRepository
-  private lateinit var profileViewModel: ProfileViewModel
+  private lateinit var savedHikesRepository: SavedHikesRepository
   private lateinit var hikesRepository: HikeRoutesRepository
   private lateinit var elevationService: ElevationService
-  private lateinit var listOfHikeRoutesViewModel: ListOfHikeRoutesViewModel
-  private val route =
-      HikeRoute(
-          id = "1",
-          bounds = Bounds(minLat = 45.9, minLon = 7.6, maxLat = 46.0, maxLon = 7.7),
-          ways = listOf(LatLong(45.9, 7.6), LatLong(45.95, 7.65), LatLong(46.0, 7.7)),
+  private lateinit var hikesViewModel: HikesViewModel
+
+  private val hikeId = "1"
+  private val detailedHike =
+      DetailedHike(
+          id = hikeId,
+          color = Hike(hikeId, true, null, null).getColor(),
+          isSaved = true,
+          plannedDate = null,
           name = "Sample Hike",
           description =
-              "A scenic trail with breathtaking views of the Matterhorn and surrounding glaciers.")
+              "A scenic trail with breathtaking views of the Matterhorn and surrounding glaciers.",
+          bounds = Bounds(minLat = 45.9, minLon = 7.6, maxLat = 46.0, maxLon = 7.7),
+          waypoints = listOf(LatLong(45.9, 7.6), LatLong(45.95, 7.65), LatLong(46.0, 7.7)),
+          elevation = listOf(0.0, 10.0, 20.0, 30.0),
+          distance = 13.543077559212616,
+          elevationGain = 68.0,
+          estimatedTime = 169.3169307105514,
+          difficulty = HikeDifficulty.DIFFICULT,
+      )
+
+  /** @param hike The hike to display on the screen. For test purposes, should always be saved. */
+  private suspend fun setupCompleteScreenWithSelected(
+      hike: DetailedHike,
+      waypointsRetrievalSucceeds: Boolean = true,
+      elevationRetrievalSucceeds: Boolean = true,
+      alreadyLoadData: Boolean = true
+  ) {
+
+    // This setup function is designed for saved hikes only. If the provided hike is not saved, it
+    // won't be loaded in the view model, hence the test will fail for sure. This message helps the
+    // developer to find out this cause without many hours of debugging.
```

Comment:
What this setup function does is :

- Load a list of saved hikes in the view model, as if the user went to the saved hikes screen
- Load (or not) its OSM data (waypoints and bounds mostly)
- Then load (or not) its elevation
- Then compute its details

The point being to have a completely loaded hike selected before the screen is loaded, so that information is displayed as expected.

However, to increase code coverage, I also checked the loading process. If the hike is not fully ready when the run hike is entered (which in practice is never the case, but code coverage...), the last details are loaded.

The first that is checked and loaded are the OSM details. The only way those are not loaded yet is for the hike to be only loaded from the saved hikes, and not from an OSM API call. Hence the choice of only accepting saved hikes. It's a rough simplification, but it shouldn't cause any issues to test only with saved hikes.

Reply:
Oh I see and it makes sense, thanks for the explanation.